transport/compress.go

```diff
@@ -0,0 +1,146 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/klauspost/compress/s2"
+	"github.com/mmatczuk/scylla-go-driver/frame"
+	"github.com/pierrec/lz4/v4"
+)
+
+var (
+	ErrComprCorrupt     = errors.New("compression: corrupt input")
+	ErrComprUnknown     = errors.New("compression: unknown algorithm")
+	errComprUnspecified = errors.New("compression: unspecified algorithm")
+)
+
+type compr struct {
+	buf    bytes.Buffer
+	tmp    bytes.Buffer
+	header []byte
+
+	codec func(read int) (written int, err error)
+	// Used in lz4 encoding
+	c lz4.Compressor
+}
+
+func (c *compr) compress(dst io.Writer, src *bytes.Buffer) (written int64, err error) {
+	c.buf = *src
+	buf := c.buf.Bytes()
+	read := len(c.buf.Bytes())
+
+	if read <= 9 {
+		nr, err := dst.Write(buf[:read])
+		return int64(nr), err
+	}
+	_ = copy(c.header, buf[:9])
+	c.header[1] |= frame.Compress
+
+	var n int
+	if n, err = c.codec(read); err != nil {
+		return 0, err
+	}
+
+	nrh, erh := dst.Write(c.header)
+	if erh != nil {
+		return int64(nrh), erh
+	}
+
+	nr, er := dst.Write(c.tmp.Bytes()[:n])
```

Comment:
How about we wrap this line with some WriteCompressed to hide tmp variable, which is used in codec function. 
What I mean is that codec function silently uses tmp, but it is not documented. I would opt for hiding it completely.

Reply:
Added documentation about tmp usage, I believe now it's ok